src/app/downloads/downloads-table.component.spec.ts

```diff
@@ -0,0 +1,142 @@
+    ['2017-09-07T00:00:00Z', 858]
+  ];
+  const podcast: PodcastModel = {
+    doc: undefined,
+    seriesId: 37800,
+    feederId: '70',
+    title: 'Pet Talks Daily'
+  };
+  const episodes: EpisodeModel[] = [
+    {
+      doc: undefined,
+      seriesId: 37800,
+      id: 123,
+      publishedAt: new Date('9/9/17'),
+      title: 'A Pet Talk Episode',
+      guid: 'abcdefg'
+    },
+    {
+      doc: undefined,
+      seriesId: 37800,
+      id: 124,
+      publishedAt: new Date(),
+      title: 'A More Recent Pet Talk Episode',
+      guid: 'gfedcba'
+    }
+  ];
+  const filter: FilterModel = {
+    podcast,
+    episodes,
+    beginDate: new Date('2017-08-27T00:00:00Z'),
+    endDate: new Date('2017-09-07T00:00:00Z'),
+    interval: INTERVAL_DAILY
+  };
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [
+        DownloadsTableComponent
+      ],
+      imports: [
+        RouterTestingModule,
+        SharedModule,
+        StoreModule.forRoot(reducers)
+      ]
+    }).compileComponents().then(() => {
+      fix = TestBed.createComponent(DownloadsTableComponent);
+      comp = fix.componentInstance;
+      fix.detectChanges();
+      de = fix.debugElement;
+      el = de.nativeElement;
+
+      // call episode and episode metrics to prime the store
+      comp.store.dispatch(new CastleFilterAction({filter}));
+      comp.store.dispatch(new CastleEpisodeMetricsAction({episode: episodes[0], filter, metricsType: 'downloads', metrics: ep0Downloads}));
+      comp.store.dispatch(new CastleEpisodeMetricsAction({episode: episodes[1], filter, metricsType: 'downloads', metrics: ep1Downloads}));
+      episodes.forEach(episode => {
+        comp.store.dispatch(new CmsEpisodeGuidAction({podcast, episode}));
+      });
+    });
+  }));
+
+  it('should transform episode data to table data', () => {
+    expect(comp.episodeTableData.length).toEqual(episodes.length);
+    expect(comp.episodeTableData[0].totalForPeriod).not.toBeNull();
+  });
+
+  it('should sort episodes by published date', () => {
+    expect(comp.episodeTableData[0].title).toEqual('A More Recent Pet Talk Episode');
+  });
+
+  it('should not show up without data to display', () => {
+    comp.episodeTableData = [];
+    expect(de.query(By.css('metrics-downloads-table'))).toBeNull();
+  });
+
+  it('should show only the episodes in filter', () => {
+    comp.store.dispatch(new CastleFilterAction({filter: {episodes: [episodes[0]]}}));
+    expect(comp.episodeTableData.length).toEqual(1);
+  });
+
+  it('should clear out episode table data when podcast changes', () => {
+    const differentPodcast: PodcastModel = {
+      doc: undefined,
+      seriesId: 37801,
+      title: 'Totally Not Pet Talks Daily'
+    };
+    comp.store.dispatch(new CastleFilterAction({filter: {podcast: differentPodcast}}));
+    expect(comp.episodeTableData.length).toEqual(0);
+  });
+
+  it('should display episode total downloads for period', () => {
```

Comment:
I feel like this test is just testing the getTotal utility function. Is there something that would cause it to not display the total here that should be tested?

Reply:
Good point. I can just check for the existence of `totalForPeriod` on each row -- what I wanted to have in the test was something that checks for the totaling having happened for every row.